Utils/azuretools-core/src/com/microsoft/azuretools/ActionConstants.java

```diff
@@ -96,11 +96,13 @@ public static ActionEntity parse(String action) {
         String START = OPERATION_PREFIX + "start";
         String STOP = OPERATION_PREFIX + "stop";
         String RESTART = OPERATION_PREFIX + "restart";
+        String SAVE = OPERATION_PREFIX + "save";
         String OPEN_IN_PORTAL = OPERATION_PREFIX + "open-in-portal";
         String SHOW_PROPERTIES = OPERATION_PREFIX + "show-properties";
         String CONNECT_TO_SERVER = OPERATION_PREFIX + "connect-to-server";
         String BIND_INTO = OPERATION_PREFIX + "bind-into";
         String LINK_TO_MODULE = OPERATION_PREFIX + "link-to-module";
+        String DO_SERVICE_LINK = OPERATION_PREFIX + "do-link";
```

Comment:
service-link-inject-env-variable

Reply:
inject env variable is more like what we do in service link before run task rather than run the before run task